src/me/john000708/barrels/Barrel.java

```diff
@@ -112,38 +111,40 @@ public void onPlace(Player player, Block block, SlimefunItem slimefunItem) {
                 if (unregisterReason.equals(UnregisterReason.EXPLODE)) {
-                    if (BlockStorage.getLocationInfo(b.getLocation(), "explosion") != null) return false;
+                    if (BlockStorage.getBlockInfo(block, "explosion") != null) return false;
                 } else if (unregisterReason.equals(UnregisterReason.PLAYER_BREAK)) {
-                    if (!BlockStorage.getLocationInfo(b.getLocation(), "owner").equals(player.getUniqueId().toString()))
+                    if (!BlockStorage.getBlockInfo(block, "owner").equals(player.getUniqueId().toString()))
                         return false;
                 }
 
-                DisplayItem.removeDisplayItem(b);
+                DisplayItem.removeDisplayItem(block);
 
-                BlockMenu inv = BlockStorage.getInventory(b);
+                BlockMenu inv = BlockStorage.getInventory(block);
 
-                if (BlockStorage.getLocationInfo(b.getLocation(), "explosion") != null)
-                    b.getWorld().dropItem(b.getLocation(), SlimefunItem.getByID("EXPLOSION_MODULE").getItem());
-                if (BlockStorage.getLocationInfo(b.getLocation(), "STRUCT_1") != null)
-                    b.getWorld().dropItem(b.getLocation(), SlimefunItem.getByID("STRUCT_UPGRADE_1").getItem());
-                if (BlockStorage.getLocationInfo(b.getLocation(), "STRUCT_2") != null)
-                    b.getWorld().dropItem(b.getLocation(), SlimefunItem.getByID("STRUCT_UPGRADE_2").getItem());
-                if (BlockStorage.getLocationInfo(b.getLocation(), "STRUCT_3") != null)
-                    b.getWorld().dropItem(b.getLocation(), SlimefunItem.getByID("STRUCT_UPGRADE_3").getItem());
-                if (BlockStorage.getLocationInfo(b.getLocation(), "protected") != null)
-                    b.getWorld().dropItem(b.getLocation(), SlimefunItem.getByID("BIO_PROTECTION").getItem());
+                if (BlockStorage.getBlockInfo(block, "explosion") != null)
+                    block.getWorld().dropItem(block.getLocation(), SlimefunItem.getByName("EXPLOSION_MODULE").getItem());
+                if (BlockStorage.getBlockInfo(block, "STRUCT_1") != null)
+                    block.getWorld().dropItem(block.getLocation(), SlimefunItem.getByName("STRUCT_UPGRADE_1").getItem());
+                if (BlockStorage.getBlockInfo(block, "STRUCT_2") != null)
+                    block.getWorld().dropItem(block.getLocation(), SlimefunItem.getByName("STRUCT_UPGRADE_2").getItem());
+                if (BlockStorage.getBlockInfo(block, "STRUCT_3") != null)
+                    block.getWorld().dropItem(block.getLocation(), SlimefunItem.getByName("STRUCT_UPGRADE_3").getItem());
+                if (BlockStorage.getBlockInfo(block, "protected") != null)
+                    block.getWorld().dropItem(block.getLocation(), SlimefunItem.getByName("BIO_PROTECTION").getItem());
```

Comment:
For all the above too:

getByName is deprecated, getByID should be used instead...